app/scripts/directives/oscSecrets.js

```diff
@@ -0,0 +1,67 @@
+"use strict";
+
+angular.module("openshiftConsole")
+
+  .directive("oscSecrets", function($uibModal, $timeout) {
+    return {
+      restrict: 'E',
+      scope: {
+        pickedSecrets: "=model",
+        allSecrets: '=',
+        namespace: "=",
+        additionalInputFieldName: "@?",
+        additionalInputFieldTooltip: "@?",
```

Comment:
It's strange to have a generic field with name and tooltip passed in. I'd much rather have the directive know what type of secret it is and just show the right thing.